docs/pkcs11.md

```diff
@@ -0,0 +1,313 @@
+# Ocicrypt Pkcs11 (Experimental)
+
+Ocicrypt supports the use of an experimental pkcs11-based protocol. This allows the ability to encrypt a container image so that it can be decrypted by a key which resides in a Hardware Security Module (HSM). In this document, we will go through a tutorial on how to setup and use this capability with a software emulated HSM, SoftHSM. 
+
+This tutorial is done on ubuntu.
+
+# Setting up SoftHSM
+
+## Requirements
+
+On top of the generic ocicrypt requirements, install the following packages:
+```
+sudo apt install -y softhsm2 p11-kit gnutls-bin
+```
+
+
+## Create SoftHSM Configuration
+
+We will setup the configurations and folders for SoftHSM. We will use the home directory to store this configuration.
+
+### Create config directories and sub directories
+```
+mkdir -p ${HOME}/.config/softhsm2/tokens
+```
+### Write configuration file
+```
+cat > ${HOME}/.config/softhsm2/softhsm2.conf <<EOF
+directories.tokendir = ${HOME}/.config/softhsm2/tokens
+objectstore.backend = file
+log.level = DEBUG
+slots.removable = false
+EOF
+```
+
+
+# Creating a pkcs11 key in SoftHSM
+
+To create a key, we need to first create a token, and then generate the private key for the token.
+
+# Create token for SoftHSM
+
+Set the configuraiton file to the one created above:
+```
```

Comment:
typo: configuration

Maybe mention that all subsequent commands need this environment variable to be set.